core/src/main/scala/org/apache/spark/deploy/master/ZookeeperSuspensionTolerantErrorPolicy.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.master
+
+import org.apache.curator.framework.state.ConnectionState
+import org.apache.curator.framework.state.ConnectionStateErrorPolicy
```

Comment:
Hi. Thank you for making a PR.
However, GitHub Action shows the compilation error. Is this compiled in your environment?
```
[ERROR] [Error] /home/runner/work/spark/spark/core/src/main/scala/org/apache/spark/deploy/master/ZookeeperSuspensionTolerantErrorPolicy.scala:21: object ConnectionStateErrorPolicy is not a member of package org.apache.curator.framework.state
318
[ERROR] [Error] /home/runner/work/spark/spark/core/src/main/scala/org/apache/spark/deploy/master/ZookeeperSuspensionTolerantErrorPolicy.scala:26: not found: type ConnectionStateErrorPolicy
319
[ERROR] [Error] /home/runner/work/spark/spark/core/src/main/scala/org/apache/spark/deploy/SparkCuratorUtil.scala:47: value connectionStateErrorPolicy is not a member of org.apache.curator.framework.CuratorFrameworkFactory.Builder
```

Reply:
Hi, 

yes this happens in my local environment now as well (I cherry-picked this from a 2.2 branch before).

The issue is caused because according to the build logs `curator-client-2.13.0.jar` is added to the core classpath by `maven-dependency-plugin:3.1.1` instead of the desired version `4.2.0`.

Strangely `./build/mvn dependency:tree` lists only the correct versions (`4.2.0`) on curator everywhere...

Reply:
Oversaw a different version in curator that was not present on the previous branch.
Compiles now.